examples/lock-app/nrfconnect/README.md

```diff
@@ -1,53 +1,101 @@
+# CHIP nRF Connect Lock Example Application
 
-An example application showing the use
-[CHIP](https://github.com/project-chip/connectedhomeip) on the Nordic nRF52840.
+The lock example provides demonstration of application running on the connected
+door lock device. It uses
+[CHIP](https://github.com/project-chip/connectedhomeip) and the nRF Connect
+platform. The example supports remote access and control of a simulated door
+lock over a low-power, 802.15.4 Thread network. It is capable of being paired
+into an existing CHIP network along with other CHIP-enabled devices. Besides of
+the remote access, application also allows controlling lock and device states
+using buttons and visualizes those states on the LEDs.
 
 <hr>
 
--   [CHIP nRF52840 Lock Example Application](#chip-nrf52840-lock-example-application)
-    -   [Introduction](#introduction)
-    -   [Device UI](#device-ui)
-    -   [Building](#building)
-        -   [Using Docker container](#using-docker-container)
-        -   [Using Native shell](#using-native-shell)
-        -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
-    -   [Configuring the example](#configuring-the-example)
-    -   [Flashing and debugging](#flashing-and-debugging)
-    -   [Accessing the command line](#accessing-the-command-line)
+-   [Overview](#overview)
+-   [Requirements](#requirements)
+-   [Device UI](#device-ui)
+-   [Building](#building)
+    -   [Using Docker container](#using-docker-container)
+    -   [Using Native shell](#using-native-shell)
+    -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
+-   [Configuring the example](#configuring-the-example)
+-   [Flashing and debugging](#flashing-and-debugging)
+-   [Accessing the command line](#accessing-the-command-line)
 
 <hr>
 
-<a name="intro"></a>
+<a name="overview"></a>
 
-## Introduction
+## Overview
 
-![nrf52840 DK](../../platform/nrfconnect/doc/images/nrf52840-dk.jpg)
+This example application is running on the nRF Connect platform, which is based
+on the
+[nRF Connect SDK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html)
+and [Zephyr RTOS](https://zephyrproject.org/). Visit CHIP's
+[nRF Connect Platform Overview](TODO:...) to read more information about
+platform structure and dependencies.
 
-The nRF52840 lock example application provides a working demonstration of a
-connected door lock device, built using CHIP, and the Nordic nRF Connect. The
-example supports remote access and control of a simulated door lock over a
-low-power, 802.15.4 Thread network. It is capable of being paired into an
-existing CHIP network along with other CHIP-enabled devices. The example targets
-the
-[Nordic nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK)
-development kit, but is readily adaptable to other nRF52840-based hardware.
+CHIP device running lock application is controlled by the CHIP controller device
```

Comment:
I wonder if it wouldn't be more helpful if all the features you mention like "test mode" etc. were separate points in the text:
1. At the first boot, the application starts with disabled Thread networking and BLE advertising.
2. Test mode...
3. QR Code Payload
...
But let others comment on that before changing anything.

Reply:
Ok, I thought that I did something similar with creating separate sections for advertising, rendezvous and provisioning, but if the first text block is still too big, I can try to create more smaller points.